From 5122f837f8d6e14971b1caac15a86990f2af26bf Mon Sep 17 00:00:00 2001
From: Kefu Chai <tchaikov@gmail.com>
Date: Fri, 3 Apr 2026 08:55:21 +0800
Subject: [PATCH] fix(python_plugins_loader): suppress -Wconversion for
 SWIG-generated code

The SWIG-generated swigpyrun.h contains an implicit size_t to int
narrowing in SWIG_Hash():

  rem -= sizeof(unsigned int);

This triggers -Werror=conversion with GCC 16 (not Clang). The bug is
fixed in SWIG master (https://github.com/swig/swig/pull/3287), but
unfortunately, not in any released SWIG version at the time of writing
(see
https://github.com/swig/swig/pull/3287/changes/296fe5e4325795e8fea933341c20b73566cffc87,
4.4.1 narrowly missed it).

This warning was previously masked by a leaked -Wconversion diagnostic
suppression in fmt < 12 (https://github.com/fmtlib/fmt/pull/4588).
fmt 12 properly scoped the suppression, exposing this pre-existing
SWIG issue.

Add -Wno-conversion to the existing warning suppression list for
the python_plugins_loader target, consistent with the existing
approach of suppressing warnings in SWIG-generated code that we
don't control.

We can revert this change once the downstream packaging picks up the
SWIG fix.

Signed-off-by: Kefu Chai <tchaikov@gmail.com>
---
 libdnf5-plugins/python_plugins_loader/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libdnf5-plugins/python_plugins_loader/CMakeLists.txt b/libdnf5-plugins/python_plugins_loader/CMakeLists.txt
index e316af7f..6275d4fb 100644
--- a/libdnf5-plugins/python_plugins_loader/CMakeLists.txt
+++ b/libdnf5-plugins/python_plugins_loader/CMakeLists.txt
@@ -28,9 +28,9 @@ add_library(python_plugins_loader MODULE python_plugins_loader.cpp)
 add_dependencies(python_plugins_loader swigpyrun-header)
 # Include directory with the gereated swigpyrun.h header
 target_include_directories(python_plugins_loader PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
-# This is need because the generated swigpyrun.h contains warnings
+# This is needed because the generated swigpyrun.h contains warnings
 # (which get treated as errors)
-target_compile_options(python_plugins_loader PRIVATE -Wno-sign-conversion -Wno-missing-field-initializers)
+target_compile_options(python_plugins_loader PRIVATE -Wno-sign-conversion -Wno-missing-field-initializers -Wno-conversion)
 
 # disable the 'lib' prefix in order to create python_plugins_loader.so
 set_target_properties(python_plugins_loader PROPERTIES PREFIX "")
-- 
2.53.0

